### PR TITLE
WIP: Beta distribution support for PPO

### LINKED ISSUE #2142: BetaDistribution policy for bounded continuous action spaces to avoid Gaussian clipping bias and improve training stability
### 🚀 Feature

Add a option of  `BetaDistribution` policy for bounded continuous action spaces to avoid Gaussian clipping bias and improve training stability.

### Motivation

[ Petrazzini & Antonelo (2021) ](https://arxiv.org/pdf/2111.02202) demonstrated that replacing Gaussian with a Beta distribution (compact support) in PPO yields significantly faster convergence, higher final rewards, and a 63 % increase in success on CarRacing-v0.

### Pitch

New `BetaDistribution` class in `stable_baselines3.common.distributions`, mirroring `DiagGaussianDistribution` with methods and optionally to use it on any `on policy algorithm`.

### Alternatives

- **Tanh-squashed Gaussian (SquashedDiagGaussianDistribution):** still biases density near ±1 and can have vanishing gradients at the tails .
- **Truncated normal**: more complex to implement and less stable under backpropagation .
- **State-Dependent Noise (gSDE)**: adapts variance but still uses Gaussian support, so does not fully eliminate bias at hard bounds .

### Additional context

_No response_

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [x] If I'm requesting a new feature, I have proposed alternatives

### PR DESCRIPTION
**Pull request by araffin**
## Description

closes https://github.com/DLR-RM/stable-baselines3/issues/2142

Note: done with zed + claude

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (`docs/misc/changelog.md`) (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)
- [ ] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by lukaskiss222** [comment]
I would recommend for beta distributions to use mean instead of mode for deterministic action by overriding get_auctions (to follow the [Petrazzini & Antonelo (2021)](https://arxiv.org/pdf/2111.02202) paper:
```python
    
    def mean(self) -> th.Tensor:
        # Mean of Beta(alpha, beta) = alpha / (alpha + beta)
        alpha = self.distribution.concentration1
        beta = self.distribution.concentration0
        return alpha / (alpha + beta)
    
    def get_actions(self, deterministic: bool = False) -> th.Tensor:
        if deterministic:
            # paper used the mean as deterministic action
            return self.mean()
        return self.sample()

```
